nilearn/mass_univariate/permuted_least_squares.py

```diff
@@ -693,6 +693,20 @@ def permuted_ols(
     else:
         intercept_test = False
 
+    # check if confounding vars contain constants
+    constants = []
+    if confounding_vars is not None:
+        for column in range(confounding_vars.shape[1]):
+            if np.unique(confounding_vars[:, column]).size == 1:
+                constants.append(column)
+        if len(constants) > 0:
+            warnings.warn(
+            'One or more "confounding_vars" are constant and will be ignored. '
+            'Tip: Use "model_intercept" (default=True) for intercept control.'
```

Comment:
Regarding the way this check is implemented, if `confounding_vars` includes an intercept term users would get the warning message whether or not `model_intercept` is True or False. So then the tip doesn't really make sense to me. Maybe it's just a matter of improving the warning.

Alternatively, if it's too difficult to control for all the possibilities we can also raise a warning to the user to check their input if `intercept_test` is a constant and `confounding_vars` has a constant term and not remove anything from `confounding_vars`. This makes it their responsibility to make sure their parameters makes sense. This is just a suggestion. Maybe there are better ways to do this.

Reply:
We should by all means avoid having duplicate regressors.

Reply:
Thanks @jrdalenberg for updating this check and the warning, it looks good now!